zerver/tornado/event_queue.py

```diff
@@ -181,6 +184,7 @@ def from_dict(cls, d: MutableMapping[str, Any]) -> "ClientDescriptor":
             d.get("stream_typing_notifications", False),
             d.get("user_settings_object", False),
             d.get("pronouns_field_type_supported", True),
+            d.get("linkifier_url_template", False),
```

Comment:
I'm not exactly sure where this needs to be, but I believe the `notify_linkifiers` code path will send new-format linkifier objects to clients that don't have the `linkifier_url_template` client capability. I believe the fix for this likely involves having any `realm_filters` events just silently dropped for clients who lack that, since the plan is we're just giving `[]` as the set of realm filters.

Reply:
OK, I got it updated so that we are dropping `realm_filters` unconditionally. For `realm_linkifiers`, `apply_event` will now check `linkifier_url_template` to decide if a state change is needed.